lib/rspec/matchers/built_in/yield.rb

```diff
@@ -98,7 +98,9 @@ def assert_valid_expect_block!
       # Not intended to be instantiated directly.
       class YieldControl < BaseMatcher
         def initialize
+          @already_set = false
           at_least(:once)
+          @already_set = false
```

Comment:
I'd prefer refactoring so that we can se the count internally and just check if the expectation is set.

Reply:
I agree, that was a mistake. I actually changed the implementation for #1168 but thought this PR had been merged already. Fixed.